src/electron/helpers/ipc/sdk/openai.ts

```diff
@@ -0,0 +1,190 @@
+import { createOpenAI } from "@ai-sdk/openai";
+import type { SDKMessage } from "@captn/react/types";
+import { APP_MESSAGE_KEY } from "@captn/utils/constants";
+import { streamText } from "ai";
+import { ipcMain, type IpcMainEvent } from "electron";
+import Store from "electron-store";
+import type { Except } from "type-fest";
+import { v4 } from "uuid";
+
+import { logError } from "@/electron/services/logger";
+import { keyStore } from "@/electron/stores";
+import type { ChatModel } from "@/shared/types/openai";
+
+export const openAIStore = new Store<Record<string, ChatModel>>({
+	name: "OPENAI_HISTORY",
+});
+
+ipcMain.on(
+	APP_MESSAGE_KEY,
+	async (
+		event: IpcMainEvent,
+		{
+			message,
+			appId,
+		}: {
+			message: SDKMessage<{
+				input?: string;
+				model?: string;
+				maxHistory?: number;
+				id: string;
+				update?: Partial<Except<ChatModel, "messages">>;
+			}>;
+			appId: string;
+		}
+	) => {
+		if (!message.action.startsWith("openai:")) {
+			return;
+		}
+
+		const channel = `${appId}:${APP_MESSAGE_KEY}`;
+		const chatId = message.payload.id;
+		let chat = openAIStore.get(chatId, {
+			label: "New Chat",
+			messages: [],
+			model: "gpt-3.5-turbo",
+		});
+
+		switch (message.action) {
```

Comment:
Can you please add a comment for each action, so that it's clear what the action is doing?